client/state/posts/selectors.js

```diff
@@ -41,6 +42,23 @@ export function getSitePost( state, siteId, postId ) {
 }
 
 /**
+ * Returns an array of post objects by site ID.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @return {?Array}         Site posts
+ */
+export function getSitePosts( state, siteId ) {
```

Comment:
>Would there be any value to providing a filter here for `type`?

Yeah, I might imagine this would live at minimum as a separate selector (`getSitePostsByType( state, siteId, type );`), or more likely using existing filtering selectors ([`getSitePostsForQuery`](https://github.com/Automattic/wp-calypso/blob/3fd70dce419cd8e6f783c799852809e006f7170a/client/state/posts/selectors.js#L79-L102)).